### ISSUE TITLE
Issues with El Capitan 10.11.2

### ISSUE THREAD

**Issue by jv-k** [syscl/M3800#3]
Hey, thanks for the reply in the other thread. What version of El Cap are you using? What are your M3800 specs?

I have the earlier model M3800 with QHD+ panel.

I followed all the steps correctly as far as I know, including the ones with Clover (I used the method suggested in this repo to install and dump ACPI, then tried your one after first boot).

Your script is nice and tidy and I've combined your script and AustinSMU's to create a fully automated build.

But one thing, how are you dealing with System Integrity Protection? Even when you disable it (I used config.plist in Clover to fully disable it), modifying system files fails. For example, the very first line of install_audio(). And the error message is hidden from the user as it's piped to a file.

The Azul and HD5000 kexts aren't loaded, nor is the ALC sound kext for me.

**Comment by syscl** [comment]
My M3800 specs:
1.Cpu: i7-4712HQ.
2. Mem: Samsung DDR3L 4G x 2 with 1600MHZ.
3. Harddisk: Sandisk XPS 960G + Samsung PM851.
4. Discrete card: HD4600(Pre-allocated Memory 128MB) + K1100M(4GB DDR5).
5. Tested bios: A07, A08, A09, A10. 
6. Audio: ALC668.
7. Display: Sharp 15.6' with 3200 x 1800.
8. Tested operation system: 10.10.x->10.11.x

Suggestions:

There's no necessary to create another script since it will cost you copious amount of time.

You can post the screenshot of the execution of the script here for further analysis, and more details are welcome(such as kextstat, boot.log, ...) !

The config.plist under my space has been modified to disable the SIP.
- If the Azul and HD5000 kexts aren't loaded correctly, you have to patch the IOKit by yourself:

sudo perl -i.bak -pe 's|\xB8\x01\x00\x00\x00\xF6\xC1\x01\x0F\x85|\x33\xC0\x90\x90\x90\x90\x90\x90\x90\xE9|sg' /System/Library/Frameworks/IOKit.framework/Versions/Current/IOKit

## sudo codesign -f -s - /System/Library/Frameworks/IOKit.framework/Versions/Current/IOKit

Once Azul and HD5000 have been loaded, the ALC sound will be loaded as well, and then run the script again to fix the lid wake problem which cost me copious amount of time to find out the solution.

Good luck buddy!

**Comment by syscl** [comment]
@johnvalai I've updated my script for further compatibility. I have tested the new script on my 10.11.3 update, and fixed some bugs that will prevent the script from patching. 

Wait for your feedback.

**Comment by jv-k** [comment]
Hey syscl, sorry for having not replied sooner, I got a little busy with work. 

I tried your updated script on a fresh copy of 10.11.3, which I later upgraded to 10.11.5, and it works pretty well, including the integrated webcam, which I had a hard time getting to work before.

A minor issue is the well known one with the audio. If you 'cold boot' into OSX after a restart, audio through the line-out jack doesn't work (but audio through the internal speakers is fine), until you either reboot by either deep sleeping, sleeping or restarting, after which it works again. If you shutdown or deep-sleep and go to Linux or Win10 and come back, the issue is still there. I understand this to be a problem of the ALC kexts, and not your script. But this is the only issue that I can find so far after a few months of usage.

Another problem is the power management on batteries, I realise this is another problem with the kexts, but the batteries drain pretty quickly.

Thanks for your work, good job on the bash scripting!
